packages/video_player/video_player/ios/Classes/FLTVideoPlayerPlugin.m

```diff
@@ -262,11 +272,20 @@ - (void)observeValueForKeyPath:(NSString*)path
       case AVPlayerItemStatusUnknown:
         break;
       case AVPlayerItemStatusReadyToPlay:
-        [item addOutput:_videoOutput];
-        [self sendInitialized];
-        [self updatePlayingState];
+        if (!_isInitialized) {
+          [item addOutput:_videoOutput];
+          [self sendInitialized];
+          [self updatePlayingState];
+        }
         break;
     }
+  } else if (context == durationContext || context == presentationSizeContext) {
```

Comment:
The issue is that when the player's ready event comes in, the player doesn't always have the presentation size or the duration available yet. So we need to defer sending the initialized call to flutter until we have all 3 (the duration, presentation size and the ready state).

Reply:
This seems to be wrong for online broadcasts. Maybe the video size is enough for us?

Reply:
Good point! I'll test this all out again with just presentation size.

Reply:
@otopba Actually, we need to keep the duration context check, because in `(void)sendInitialized {` there is already the following check:

```
// The player may be initialized but still needs to determine the duration.
    if ([self duration] == 0) {
      return;
    }
```
So, the issue is that we don't initialize, and then nothing causes us to send the initialization if the duration is the last thing to get notified on the player item.